py/mpprint.c

```diff
@@ -499,10 +499,11 @@ int mp_vprintf(const mp_print_t *print, const char *fmt, va_list args) {
                     break;
                 }
                 #endif
-                if (prec < 0) {
-                    prec = strlen(str);
+                size_t len = strlen(str);
```

Comment:
It's a shame that `strlen()` is now executed unconditionally, since it takes O(N) time.  But I don't really see any way around that.